app/controllers/vacancies_controller.rb

```diff
@@ -34,6 +40,16 @@ def show
 
   private
 
+  UUID_REGEX = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/
+
+  def canonical
+    return unless SEO_CANONICAL_TEST_VACANCIES.include?(vacancy.id)
+
+    return unless UUID_REGEX.match(id) || !job_path(vacancy).end_with?(vacancy.slug)
```

Comment:
Having slept on it, I'm not sure how I feel about this - are we overthinking this? Given that we redirect _any request_ to the jobs path to `job_url(vacancy)`, I reckon this could just be returning `job_url(vacancy)` in any case - for now just for the two test vacancies.

Reply:
i.e. in `VacanciesHelper`:

```ruby
def canonical_url(vacancy)
  return nil unless SEO_CANONICAL_TEST_VACANCIES.include?(vacancy.id)
  
  job_url(vacancy)
end
```